cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -39,6 +39,46 @@
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
+    private URI addCookieInterceptorIfRequired(URI uri) {
+        String urlProtocol = uri.getScheme();
+        String urlHost = uri.getHost();
+        String urlPath = uri.getRawPath();
+
+        //Check if port exists
+        int urlPort = uri.getPort();
+        if (urlPort < 0) {
+            if (urlProtocol.equals("http")) {
```

Comment:
Oh I see, I looked at the doc for `URL` not `URI` dam variable names, I think I'll change them while i'm there